lib/stat_tracker.rb

```diff
@@ -208,13 +196,92 @@ def most_tackles(season)
   end
 
   def fewest_tackles(season)
-    season_games = @games.find_all{|game| game.season == season}
-    season_game_ids = season_games.map{|game| game.game_id}
-    team_performances = @game_teams.find_all{|team| season_game_ids.include?(team.game_id)}
+    team_performances = team_games_by_season(season)
     performance_by_team = team_performances.group_by{|team| team.team_id}
     team_tackles = performance_by_team.transform_values do |team|
       team.sum {|game| game.tackles}
     end
     @teams.find {|team| team.team_id == team_tackles.min_by {|team, tack| tack}[0]}.team_name
   end
+
+  def team_info(team_id)
+    # finds a specific team via their id
+     team = team_by_id(team_id)
+     # returns an array of the team object's instance variables, then iterates
+     # over that array, deletes the '@' from the front of the instance variable
+     # and assigns that as a key, then sets the value equal to the key by again
+     # removing the '@' and then passing that as a method call then returning it
+     # all as a hash
+      team_data = team.instance_variables.map { |key,value| ["#{key}".delete("@"), value = team.send("#{key}".delete("@").to_sym)]}.to_h
+      # searches the hash for a key, value pair whose key is "stadium" then deletes it.
+      team_data.delete_if {|k,v| k == "stadium"}
+  end
+
+  def best_season(team_id)
+    team_seasons = @games.find_all {|game| game.home_team_id  == team_id || game.away_team_id == team_id }
+    seasons = team_seasons.group_by {|game| game.season}
+    seasons.each do |season, season_games|
+      season_game_ids = season_games.map{|game| game.game_id}
+      team_games = @game_teams.find_all { |game| game.team_id == team_id && season_game_ids.include?(game.game_id)}
+      seasons[season] = calculate_win_percentage(team_games)
+    end
+    seasons.max_by { |season, win_pct|
+      win_pct }[0]
+  end
+
+  def worst_season(team_id)
+    team_seasons = @games.find_all {|game| game.home_team_id  == team_id || game.away_team_id == team_id }
+    seasons = team_seasons.group_by {|game| game.season}
+    seasons.each do |season, season_games|
+      season_game_ids = season_games.map{|game| game.game_id}
+      team_games = @game_teams.find_all { |game| game.team_id == team_id && season_game_ids.include?(game.game_id)}
+      seasons[season] = calculate_win_percentage(team_games)
+    end
+    seasons.min_by { |season, win_pct|
+      win_pct }[0]
```

Comment:
Great job getting these to work guys. Let's refactor them into helper methods when we get the chance.